website/source/docs/platform/k8s/mesh-gateways.html.md

```diff
@@ -0,0 +1,360 @@
+description: |-
+    This documentation describes how to configure Connect Mesh Gateways in Kubernetes.
+---
+
+# Mesh Gateways on Kubernetes <sup>(beta)</sup>
+
+
+Mesh gateways enable routing of Connect traffic between different Consul datacenters.
+See [Mesh Gateways](/docs/connect/mesh_gateway.html) for more details.
+
+To deploy Gateways in Kubernetes, use our [consul-helm](/docs/platform/k8s/helm.html) chart.
+
+## Usage
+There are many configuration options for gateways nested under the [`meshGateway`](/docs/platform/k8s/helm.html#v-meshgateway) key
+of the consul-helm chart:
+
+```yaml
+meshGateway:
+  enabled: true
+```
+
+See the [chart reference](/docs/platform/k8s/helm.html#v-meshgateway) for full documentation
+on all the options.
+
+## Architecture
+Gateways run as `Pod`s in a `Deployment`. They forward requests from other datacenters
+through to the correct services running in Kubernetes.
+
+Since gateways need to be accessed from other datacenters they will need to be exposed. The
+recommended way to do this is via a load balancer.
+
+### Prequisites
+The following conditions must be satisfied for gateways to work between Consul datacenters
+in Kubernetes:
+
+1. Both clusters are running Consul **v1.6.0-beta1** since gateways are a beta feature.
+    This can be set via the helm chart:
+    
+    ```yaml
+    global:
+      image: consul:1.6.0-beta1
+    ```
+1. Each cluster has a unique datacenter name, ex. `dc1` and `dc2`.
+    This can be set via the helm chart:
+    
+    ```yaml
+    global:
+      datacenter: dc2 # use dc1 in your other datacenter.
+    ``` 
+    
+1. The Consul datacenters are joined. This will require that each Consul server agent
+   is accessible from each datacenter's server agents and that the servers are
+   configured to join with the other datacenter. See our [Multiple Datacenters Guide](https://learn.hashicorp.com/consul/security-networking/datacenters).
+1. The `primary_datacenter` setting is set to a single datacenter. This is required
+   for Connect requests to work across datacenters. This can be set via the helm chart:
+   
+    ```yaml
+    client:
+      extraConfig: |
+        {
+          "primary_datacenter": "dc1"
+        }
+    server:
+      extraConfig: |
+        {
+          "primary_datacenter": "dc1"
+        }
+    ```
+1. If using Kubernetes, Connect injection must be enabled. This can be set via the helm chart:
+
+    ```yaml
+    connectInject:
+      enabled: true
+    ```
+1. gRPC must be enabled. This can be set via the helm chart:
+
+    ```yaml
+    client:
+      grpc: true
+    ```
+1. If setting a global default [gateway mode](/docs/connect/mesh_gateway.html#modes-of-operation),
+   central config must be enabled for all agents so that they actually use the default mode.
+   This can be set via the helm chart:
+
+    ```yaml
+    connectInject:
+      centralConfig:
+       enabled: true
+    ```
+
+## Guide to Using Mesh Gateways To Connect Two Kubernetes Clusters
```

Comment:
Yes I do. I think its eventual home should be there, but I'm worried that it's not complete enough to live there in its current state. I think having at least these docs will be useful until we get a proper guide together but am happy to hear otherwise.